daml/splice-amulet/daml/Splice/AmuletRules.daml

```diff
@@ -1138,28 +1142,26 @@ data AppTransferContext = AppTransferContext
     featuredAppRight : Optional (ContractId FeaturedAppRight)
   deriving (Show, Eq)
 
-exerciseAppTransfer : AppTransferContext -> Transfer -> Update TransferResult
-exerciseAppTransfer AppTransferContext{..} transfer =
+exerciseAppTransfer : Party -> AppTransferContext -> Transfer -> Update TransferResult
+exerciseAppTransfer expectedDso AppTransferContext{..} transfer = do
+  -- prudent engineering: check that the expected DSO matches the one of the AmuletRules used for the transfer.
+  _ <- fetchPublicReferenceData (ForDso with dso = expectedDso) amuletRules (AmuletRules_Fetch with p = transfer.sender)
```

Comment:
@moritzkiefer-da : I've encountered some problems with tx-log parsing due to these new nodes, and noticed that their result is (obviously) really large.

I'm a bit worried about the increase in traffic from this change. It seems better to me to invest in the more principled change where we have `AmuletRules_Transfer` accept and check an `expectedDso` field.

I don't think we should land this though before today's release.

wdyt?

Reply:
Good catch, let's postpone it.

Reply:
OK. I'll adjust this PR to only add the comments.